src/lib/services/graphql/graphql-subscriptions-jsdom-only.spec.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @jest-environment jsdom
+ */
+import { WebSocket } from 'mock-socket';
+import { NEVER, Subject, firstValueFrom } from 'rxjs';
+import { GraphQLSubscriptionsFixture } from './graphql-subscriptions-fixture';
+import { sleepMs } from '../../util/sleep-ms';
+import { ConnectionStatus } from '../../model/connection-status';
+
+jest.mock('isomorphic-ws', () => WebSocket);
+jest.mock('../../util/sleep-ms', () => ({
+  sleepMs: jest.fn(),
+}));
+
+/**
+ * The message sequence during these tests is the following:
+ *
+ * client -> server: 'connection_init'
+ * server -> client: 'connection_ack'
+ * client -> server: 'start', ID, query
+ * server -> client: 'data', ID, payload
+ * client -> server: 'end', ID
+ * ...
+ */
+
+// Close codes: https://www.rfc-editor.org/rfc/rfc6455#section-7.4
+describe('GraphQL subscriptions', () => {
+  let fixture: GraphQLSubscriptionsFixture;
+
+  beforeEach(async () => {
+    fixture = new GraphQLSubscriptionsFixture();
+  });
+
+  afterEach(async () => {
+    await fixture.cleanup();
+  });
+
+  it('when the browser returns online before a retry attempt, it will reconnect sooner', async () => {
```

Comment:
Proposed rename after going through the code execution after not really getting what the test does

`should reconnect based on ping-pong before "onclose" retries connection`